std/math.d

```diff
@@ -1417,10 +1433,28 @@ real sinh(real x) @safe pure nothrow @nogc
 }
 
 /// ditto
-double sinh(double x) @safe pure nothrow @nogc { return sinh(cast(real) x); }
+double sinh(double x) @safe pure nothrow @nogc
+{
+    if (fabs(x) > double.mant_dig * LN2)
```

Comment:
`LN2` is of type `real`, so better cast to double to prevent the comparison being performed in `real` precision.